docs/content/sips/xxx-manifest-redesign.md

```diff
@@ -0,0 +1,123 @@
+title = "SIP xxx - Application Manifest Redesign"
+template = "main"
+date = "2022-05-20T13:22:30Z"
+---
+
+Summary: A new design for Application Manifests (`spin.toml`)
+
+Owner: lann.martin@fermyon.com
+
+Created: May 20, 2022
+
+## Background
+
+Spin's Application Manifest file (usually named `spin.toml`) describes a Spin application's metadata,
+components, and trigger configurations. The current design supports a single trigger type and combines
+component and trigger configuration with an implicit one-to-one relationship:
+
+```toml
+name = "hello-world"
+trigger = { type = "http", base = "/hello" }
+
+[[component]]
+id = "world"
+# ...
+[component.trigger]
+route = "/world"
+```
+
+In order to support future features, we would like to enable more flexible trigger configurations:
+
+- Multiple trigger types in an application
+- Multiple triggers associated with a single component
```

Comment:
Isn't the difference of trigger function signature prevents one component associating with multiple triggers? 

For example, a Redis trigger and a HTTP trigger can't associae with a single component since their require linking to different .wit files.

Reply:
A component can implement multiple interfaces.

Reply:
Theoretically, you can do that today if you implement and export more than one interface (both the HTTP and the Redis entrypoint) and use the same `.wasm` file.